### DISCUSSION TITLE
Repository import fails

### DISCUSSION THREAD

**Discussion by stevelhanna** [community/community#78072] [New to GitHub]
### Select Topic Area

Question

### Body

I'm an open source newbie and trying this for the first time.  Trying to import the Odoo/odoo public repository on github into my own private repository so I can do some development.  It runs for about a couple of hours then I get an email saying that it failed.  The is no log referenced in the email.  I have told it to try again but it failed a second time.  Have no idea where to look for diagnostic information.  How do I troubleshoot this?

**Comment by erickdc7** [comment]
When troubleshooting a repository import failure on GitHub, here are some steps you can take to diagnose and resolve the issue:

1. **Check GitHub's Documentation**: Review GitHub's documentation on [troubleshooting cloning errors](https://docs.github.com/en/repositories/creating-and-managing-repositories/troubleshooting-cloning-errors) for common issues and solutions.
2. **Verify Repository Size**: Ensure the repository you're importing isn't too large. [GitHub has a recommended size limit for repositories, and importing very large repositories can cause issues](https://docs.github.com/en/repositories/creating-and-managing-repositories/troubleshooting-cloning-errors).
3. **Check Access Permissions**: Make sure you have the necessary permissions to access the repository you're trying to import. If it's a private repository, you'll need to have access rights.
4. **Use Personal Access Tokens**: If authentication is required, use a personal access token instead of your password. This is especially important if you have two-factor authentication enabled.
5. **Check Network Issues**: Network problems can interrupt the import process. Ensure you have a stable internet connection during the import.
6. **Contact GitHub Support**: If you're unable to resolve the issue using the above steps, reach out to GitHub support for assistance. Provide them with details about the repository and the steps you've taken so far.

Remember to provide as much information as possible when seeking help, including any error messages you've received and the actions you've tried. This will help support staff or the community to assist you more effectively.

**Comment by BhathiyaPrasad** [comment]
I also had the same error , what I am suggesting you is most of the times it occurs due to low speed of the network connection, best thing is try to download it as a zip version or do a ssh local setup, or find a better network

**Comment by DsMans0021** [comment]
The Odoo repository is very large, with a long commit history, many branches, and thousands of files, which often causes GitHub’s Importer to fail due to timeouts or memory limits. This happens because the importer service on GitHub has resource constraints and can struggle with projects of this scale. The failure emails you receive do not contain logs because GitHub does not provide public access to its internal import logs. To work around this, you can bypass the importer entirely by manually cloning and pushing the repository. First, clone Odoo locally using the mirror option to preserve all branches, tags, and commit history. After cloning, create an empty private repository on your GitHub account and push the mirrored content to it. This process avoids GitHub’s importer timeout and gives you full control over the transfer. If you do not need the complete history, you can perform a shallow clone with a limited commit depth to reduce the size and speed up the operation, then push only the branch you need. Using SSH instead of HTTPS can provide a more stable connection, especially on slower or less reliable networks. Make sure you are using a recent version of Git, as newer versions handle large repositories more efficiently. If you encounter persistent network interruptions, try disabling any firewall or antivirus temporarily. As a last resort, you can download the repository as a ZIP file from GitHub and initialize it as a new repository, though this method will not preserve the commit history.

**Comment by nairan660** [comment]
add conquist

**Comment by queenofcorgis** [comment]
@stevelhanna did anyone of the answers solve this? I also recommend for large repositories to clone and push manually.

**Comment by piyush-pooh** [comment]
Why the GitHub import fails
The GitHub “Import repository” feature often fails for very large repositories like Odoo/odoo. The importer has time and size limits and provides no detailed logs or diagnostics, so failures are usually silent and repeatable.
How to troubleshoot and fix it
There are no logs available for failed imports. The recommended and reliable approach is to clone locally and push manually.
Option 1: Full history (recommended)
git clone --mirror https://github.com/odoo/odoo.git
cd odoo.git
git remote set-url origin https://github.com/<your-username>/<private-repo>.git
git push --mirror
Option 2: Latest code only (faster)
git clone --depth=1 https://github.com/odoo/odoo.git
cd odoo
git remote add origin https://github.com/<your-username>/<private-repo>.git
git push -u origin main
